node/src/Pos/Delegation/Logic/VAR.hs

```diff
@@ -419,20 +419,42 @@ dlgVerifyBlocks blocks = do
             (revokeIssuers, changeIssuers) =
                 bimap toIssuers toIssuers $ partition isRevokePsk proxySKs
 
-        -- Check 3: Issuers have enough money (though it's free to revoke).
-        when (any (not . (`HS.member` richmen) . addressHash) changeIssuers) $
-            throwError $ sformat ("Block "%build%" contains psk issuers that "%
-                                  "don't have enough stake")
-                                 (headerHash blk)
-
-        -- Check 4: No issuer has posted psk this epoch before.
+        forM_ proxySKs $ \psk -> do
+            let stakeholderId = addressHash $ pskIssuerPk psk
+            let iPk = pskIssuerPk psk
+            let dPk = pskDelegatePk psk
+
+            -- Check 3: Issuers have enough money (though it's free to revoke).
+            when (not (isRevokePsk psk) &&
+                  not (stakeholderId `HS.member` richmen)) $
+                throwError $ sformat
+                    ("Block "%build%" contains psk issuers that "%
+                    "don't have enough stake: "%build)
+                    (headerHash blk) psk
+
+            -- Check 4: There are no psks that duplicate previous ones
+            -- compared by (i,d). That is, if i delegated to d using psk1,
+            -- we forbid using psk2 (in the next epoch) if psk2 delegates
```

Comment:
What does "in the new epoch" mean here? As I said, the issue is reproducible within a single epoch.